### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by TomFanella4**
# Changes
1. Add Toast functionality to display helpful notifications to the user
1. Implemented notifications for:
    - Login
    - Logout
    - Profile Creating Field Sanitation
    - Settings Saved
1. Refactor Login action

### CODE REVIEW

File: client/app/actions/index.js [hunk 1]
```diff
@@ -90,6 +94,32 @@ export const modifyUserProfile = userProfile => ({
...
+      const uri = `https://meetover.herokuapp.com/login/${result.params.code}` +
+        `?redirect_uri=${encodeURIComponent(redirectUri)}`;
+      const init = { method: 'POST' };
+
+      const response = await fetch(uri, init);
+      const { profile, token, firebaseCustomToken } = await response.json();
+      const firebaseIdToken = await fetchIdToken(firebaseCustomToken)
+        .catch(err => null);
+
+      dispatch(createProfile({ ...profile, token, firebaseCustomToken }));
```
Comment (mpace965) [inline client/app/actions/index.js]: We should be passing `firebaseIdToken` here.

File: client/app/reducers/navigation.js [hunk 2]
```diff
@@ -60,14 +61,22 @@ export default navigation = (state = initialState, action) => {
...
+          textStyle: { fontFamily: 'pt-sans' },
+        });
+      }
+      // return {
+      //   ...state,
+      //   stateForLoggedOut: AppNavigator.router.getStateForAction(
+      //     NavigationActions.back(),
+      //     stateForLoggedOut
+      //   )
+      // };
```
Comment (mpace965) [inline client/app/reducers/navigation.js:79]: Should this commented code stay?

File: client/app/screens/CreateProfileScreen.js [hunk 3]
```diff
@@ -95,7 +96,31 @@ class CreateProfileScreen extends React.Component {
...
+        duration: 3000,
+        textStyle: { fontFamily: 'pt-sans' },
+      });
+      return;
+    }
+
     saveProfileAndLoginAsync({ ...userProfile, isAuthenticated: true });
+    Toast.show({
+      text: `${userProfile.firstName} Signed In`,
+      textStyle: { fontFamily: 'pt-sans' }
```
Comment (mpace965) [inline client/app/screens/CreateProfileScreen.js]: What do you think about having a toast constant object that we can extend so we don't need to write `textStyle` every time?

Eg: constants.js
```js
export const toastDefaults = { textStyle: { fontFamily: 'pt-sans' } };
```

screen.js:
```js
Toast.show({ ...toastDefaults, text: 'blah' });
```

**Review by mpace965 — DISMISSED** [review]
Code looks good. A few minor changes requested. Also, works on my device.

File: client/app/reducers/navigation.js (hunk 2 above)
Comment (mpace965) [reply client/app/reducers/navigation.js:79]: Adding a `// TODO`

Comment (TomFanella4) [reply client/app/reducers/navigation.js:79]: Done

**Pull request merged**